src/controllers/itineraryItem.js

```diff
@@ -0,0 +1,102 @@
+const ItineraryItem = require('../models/itineraryItem');
+const { StatusCodes } = require('http-status-codes');
+const { BadRequestError } = require('../errors/bad_request');
+const { NotFoundError } = require('../errors/not_found');
+
+const getAllItineraryItems = async (req, res) => {
+    const itineraryItems = await ItineraryItem.find({ createdBy: req.user.userId }).sort('createdAt');
+    console.log('Fetched Itinerary Items:', itineraryItems);
+    res.status(StatusCodes.OK).json({ itineraryItems, count: itineraryItems.length });
+}
+
```

Comment:
do we want to sort by the datetime the record was 'createdAt' or by the date that the event will take place?